oauthproxy_test.go

```diff
@@ -1678,6 +1678,11 @@ func TestGetRedirect(t *testing.T) {
 			url:              "/foo/bar",
 			expectedRedirect: "/foo/bar",
 		},
+		{
```

Comment:
Can you add a fragment URI test to the table as well?

Reply:
What's the expected output here? Fragments aren't sent by a browser as part of a request so I don't think a redirect URL could ever contain one could it?

Reply:
At first I actually used a fragment, but I've found in my testing that the fragment is reapplied by the browser (chrome) after redirect. This doc seems to indicate Fragment is stripped, as @JoelSpeed mentioned: https://golang.org/pkg/net/url/#ParseRequestURI

At first I had a fragment in this test case, but then I needed this change to add fragment because URL.RequestURI() stripped it:
```
diff --git a/oauthproxy.go b/oauthproxy.go
index 4b310b9..5fa2572 100644
--- a/oauthproxy.go
+++ b/oauthproxy.go
@@ -580,7 +580,12 @@ func (p *OAuthProxy) GetRedirect(req *http.Request) (redirect string, err error)
                redirect = req.Form.Get("rd")
        }
        if !p.IsValidRedirect(redirect) {
-               redirect = req.URL.Path
+               // Use RequestURI to preserve ?query
+               redirect = req.URL.RequestURI()
+               // RequestURI() will not provide a fragment if present, so grab that here
+               if req.URL.Fragment != "" {
+                       redirect += "#" + req.URL.Fragment
+               }
                if strings.HasPrefix(redirect, p.ProxyPrefix) {
                        redirect = "/"
                }
```

I think this code would only be needed to satisfy tests, but not really used at runtime. Let me know what you think.

Reply:
Intestering.  I didn't realize that's how it worked under the hood. TIL 😄